fsharp-backend/src/LibExecution/StdLib/LibList.fs

```diff
@@ -962,123 +963,97 @@ let fns : List<BuiltInFn> =
-//               let result =
-//                 (* We reverse here so that the [foldi] consing happens in the correct order.
-//                 * It does mean that the index passed by [foldi] counts from the end *)
-//                 l |> List.rev |> List.foldi (Ok ([], [])) fold_fn
-//               in
-//               ( match result with
-//               | Ok (res_a, res_b) ->
-//                   DList [DList res_a; DList res_b]
-//               | Error v ->
-//                   v )
-//           | _ ->
-//               incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//     ; previewable = Pure
-//     ; deprecated = NotDeprecated }
+    { name = fn "List" "zipShortest" 0
+      parameters = [ Param.make "as" (TList varA) "";  Param.make "bs" (TList varB) "" ]
+      returnType = TList varA
+      description = "Returns a list of parallel pairs from `as` and `bs`.
+        If the lists differ in length, values from the longer list are dropped.
+        For example, if `as` is `[1,2]` and `bs` is `[\"x\",\"y\",\"z\"]`, returns `[[1,\"x\"], [2,\"y\"]]`.
+        Use `List::zip` if you want to enforce equivalent lengths for `as` and `bs`.
+        See `List::unzip` if you want to deconstruct the result into `as` and `bs` again."
+      fn =
+        (function
+        | state, [ DList l1; DList l2 ] ->
+            // We have to do this munging because OCaml's map2
+            // and Fsharp's zip enforces lists of the same length
+            let len = min (bigint (List.length l1)) (bigint (List.length l2))
+            let l1 = List.take (int len) l1
+            let l2 = List.take (int len) l2
+            List.zip l1 l2
+            |> List.map (fun (val1, val2) -> DList [ val1; val2 ])
+            |> DList
+            |> Value
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+    { name = fn "List" "zip" 0
+      parameters = [ Param.make "as" (TList varA) "";  Param.make "bs" (TList varB) "" ]
+      returnType = TOption(TList(TList varA))
+      description =
+        "If the lists have the same length, returns `Just list` formed from parallel pairs in `as` and `bs`.
+        For example, if `as` is `[1,2,3]` and `bs` is `[\"x\",\"y\",\"z\"]`, returns `[[1,\"x\"], [2,\"y\"], [3,\"z\"]]`.
+        See `List::unzip` if you want to deconstruct `list` into `as` and `bs` again.
+        If the lists differ in length, returns `Nothing` (consider `List::zipShortest` if you want to drop values from the longer list instead)."
+      fn =
+        (function
+        | state, [ DList l1; DList l2 ] ->
+            let result =
+              if (bigint (List.length l1) <> bigint (List.length l2)) then
+                None
+              else
+                List.zip l1 l2
+                |> List.map (fun (val1, val2) -> DList [ val1; val2 ])
+                |> DList
+                |> Some
+
+            match result with
```

Comment:
I think you could combine this with the previous statement and it would be clearer